lib/python/ray/worker.py

```diff
@@ -102,6 +102,102 @@ def __del__(self):
     """Deallocate the relevant segment to avoid a memory leak."""
     ray.lib.unmap_object(self.handle, self.segmentid)
 
+class RayReusables(object):
+  """An object used to store Python variables that are shared between tasks.
+
+  Each worker process will have a single RayReusables object. This class serves
+  two purposes. First, some objects are not serializable, and so the code that
+  creates those objects must be run on the worker that uses them. This class is
+  responsible for running the code that creates those objects. Second, some of
+  these objects are expensive to create, and so they should be shared between
+  tasks. However, if a task mutates a variable that is shared between tasks,
+  then the behavior of the overall program may be nondeterministic (it could
+  depend on scheduling decisions). To fix this, if a task uses a one of these
+  shared objects, then tht shared object will be reinitialized after the task
+  finishes. Since the initialization may be expensive, the user can pass in
+  custom reinitialization code that resets the state of the shared variable to
+  the way it was after initialization. If the reinitialization code does not do
+  this, then the behavior of the overall program is undefined.
+
+  Attributes:
+    _names (List[str]): A list of the names of all the reusable variables.
+    _initializers (dict[str, [Callable[[], object]])]: A dictionary mapping the
+      names of the reusable variables to the code for initializing them.
+    _reinitializers (Dict[str, Callable[[object], object]]): A dictionary
+      mapping the names of the reusable variables to the code for reinitializing
+      them. For reusable variables for which reinitializer code is not provided,
+      the reinitializer here essentially wraps the initializer.
+    _used (List[str]): A list of the names of all the reusable variables that
+      have been accessed within the scope of the current task. This is reset to
+      the empty list after each task.
+  """
+  def __init__(self):
+    """Initialize a RayReusables object."""
+    self._names = []
+    self._initializers = {}
+    self._reinitializers = {}
+    self._used = []
+
+  def _reinitialize(self):
+    """Reinitialize the reusable variables that the current task used."""
+    for name in self._used:
+      current_value = self.__getattribute__(name)
```

Comment:
In that case I think you should use the former if possible, since I think the latter is meant to be defined but not called directly.